build.assets/tooling/cmd/notarize-apple-binaries/hashicorpLoggerAdapter.go

```diff
@@ -0,0 +1,228 @@
+// are used to elide expensive logging code based on the current level.
+func (hclla HCLogLogrusAdapter) IsTrace() bool {
+	return hclla.logger.IsLevelEnabled(logrus.TraceLevel)
+}
+
+// Indicate if DEBUG logs would be emitted. This and the other Is* guards
+func (hclla HCLogLogrusAdapter) IsDebug() bool {
+	return hclla.logger.IsLevelEnabled(logrus.DebugLevel)
+}
+
+// Indicate if INFO logs would be emitted. This and the other Is* guards
+func (hclla HCLogLogrusAdapter) IsInfo() bool {
+	return hclla.logger.IsLevelEnabled(logrus.InfoLevel)
+}
+
+// Indicate if WARN logs would be emitted. This and the other Is* guards
+func (hclla HCLogLogrusAdapter) IsWarn() bool {
+	return hclla.logger.IsLevelEnabled(logrus.WarnLevel)
+}
+
+// Indicate if ERROR logs would be emitted. This and the other Is* guards
+func (hclla HCLogLogrusAdapter) IsError() bool {
+	return hclla.logger.IsLevelEnabled(logrus.ErrorLevel)
+}
+
+// ImpliedArgs returns With key/value pairs
+func (hclla HCLogLogrusAdapter) ImpliedArgs() []interface{} {
+	return hclla.impliedArgs
+}
+
+// Creates a sublogger that will always have the given key/value pairs
+func (HCLogLogrusAdapter) With(args ...interface{}) hclog.Logger {
+	// Ensure that there is a key for every value
+	if len(args)%2 != 0 {
+		extraValue := args[len(args)-1]
+		// Pulled from https://github.com/hashicorp/go-hclog/blob/main/intlogger.go#L731
+		args[len(args)-1] = "EXTRA_VALUE_AT_END"
+		args = append(args, extraValue)
+	}
+
+	newLogger := NewHCLogLogrusAdapter()
+	newLogger.impliedArgs = append(newLogger.impliedArgs, args)
+	return newLogger
+}
+
+// Returns the Name of the logger
+func (hclla HCLogLogrusAdapter) Name() string {
+	return hclla.name
+}
+
+// Create a logger that will prepend the name string on the front of all messages.
+// If the logger already has a name, the new value will be appended to the current
+// name. That way, a major subsystem can use this to decorate all it's own logs
+// without losing context.
+func (hclla HCLogLogrusAdapter) Named(name string) hclog.Logger {
+	if hclla.name != "" {
+		name = fmt.Sprintf("%s.%s", name, hclla.name)
+	}
+
+	return hclla.ResetNamed(name)
+}
+
+// Create a logger that will prepend the name string on the front of all messages.
+// This sets the name of the logger to the value directly, unlike Named which honor
+// the current name as well.
+func (HCLogLogrusAdapter) ResetNamed(name string) hclog.Logger {
+	newLogger := NewHCLogLogrusAdapter()
+	newLogger.name = name
+
+	return newLogger
+}
+
+// Updates the level. This should affect all related loggers as well,
+// unless they were created with IndependentLevels. If an
+// implementation cannot update the level on the fly, it should no-op.
+func (hclla HCLogLogrusAdapter) SetLevel(level hclog.Level) {
+	hclla.logger.SetLevel(hclogLevelToLogrusLevel(level))
+}
+
+// Returns the current level
+func (hclla HCLogLogrusAdapter) GetLevel() hclog.Level {
+	switch hclla.logger.GetLevel() {
+	case logrus.FatalLevel:
```

Comment:
`case` doesn't fall through in Go like you may be accustomed to. As a result, fatal and panic will actually get converted to `NoLevel` here.

To fix, you can use the `fallthrough` keyword, or better yet, combine them into the same case:

```
switch hclla.logger.GetLevel() {
case logrus.FatalLevel, logrus.PanicLevel, logrus.ErrorLevel:
// ...
```

Reply:
**Thank you**! I definitely missed this. Updated